tiledb/api/c_api/array_schema/array_schema_api.cc

```diff
@@ -181,6 +182,54 @@ capi_return_t tiledb_array_schema_timestamp_range(
   return TILEDB_OK;
 }
 
+capi_return_t tiledb_array_schema_get_enumeration_from_name(
+    tiledb_ctx_t* ctx,
+    tiledb_array_schema_t* array_schema,
+    const char* enumeration_name,
+    tiledb_enumeration_t** enumeration) {
+  ensure_array_schema_is_valid(array_schema);
+  ensure_output_pointer_is_valid(enumeration);
+
+  if (enumeration_name == nullptr) {
+    throw CAPIException("'enumeration_name' must not be null");
+  }
+
+  array_schema->load_enumeration(ctx, enumeration_name);
+
+  auto ptr = array_schema->get_enumeration(enumeration_name);
+  *enumeration = tiledb_enumeration_handle_t::make_handle(ptr);
+
+  return TILEDB_OK;
+}
+
+capi_return_t tiledb_array_schema_get_enumeration_from_attribute_name(
+    tiledb_ctx_t* ctx,
+    tiledb_array_schema_t* array_schema,
+    const char* attribute_name,
+    tiledb_enumeration_t** enumeration) {
+  ensure_array_schema_is_valid(array_schema);
+  ensure_output_pointer_is_valid(enumeration);
+
+  tiledb_attribute_t* attribute;
+  capi_return_t getattr = tiledb_array_schema_get_attribute_from_name(
```

Comment:
I chose to do this so that if the attribute were missing we would get the same error message.  But if this is not something we are supposed to do then I'll change it.

As for the leak - oops!  Thanks for catching that

Reply:
@rroelke Can we please fix that before it goes into released code? We are planning a 2.27 in the upcoming week. Thanks!

Reply:
https://github.com/TileDB-Inc/TileDB/pull/5368/commits/bcaa3e5d621ef8500a6051a76645c7126297d556